x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtAuthenticator.java

```diff
@@ -170,7 +179,64 @@ private List<JwtStringClaimValidator> getRequireClaimsValidators() {
         final Settings requiredClaims = realmConfig.getSetting(JwtRealmSettings.REQUIRED_CLAIMS);
         return requiredClaims.names().stream().map(name -> {
             final List<String> allowedValues = requiredClaims.getAsList(name);
-            return new JwtStringClaimValidator(name, allowedValues, false);
+            return new JwtStringClaimValidator(name, false, allowedValues, List.of());
         }).toList();
     }
+
+    private static JwtStringClaimValidator getSubjectClaimValidator(
+        RealmConfig realmConfig,
+        @Nullable Map<String, String> fallbackClaimLookup
+    ) {
+        validateAllowedSubjectsSettings(realmConfig);
+        return new JwtStringClaimValidator(
+            "sub",
+            true,
+            fallbackClaimLookup,
+            realmConfig.getSetting(JwtRealmSettings.ALLOWED_SUBJECTS),
+            realmConfig.getSetting(JwtRealmSettings.ALLOWED_SUBJECT_PATTERNS)
+        );
+    }
+
+    private static void validateAllowedSubjectsSettings(RealmConfig realmConfig) {
```

Comment:
Good point. I've pushed [f095252](https://github.com/elastic/elasticsearch/pull/102426/commits/f09525221cf531e80049bbe47fb696ff537153b9) , but the "richness" of the error description decreased because it's not possible to tell if the dependent setting (of some other setting) is specified or not (validation has access to the default value in case it's not specified).
So now the error message only tells you that one of `allowed_subjects` or `allowed_subject_patterns` must be specified and not be empty.